### DISCUSSION TITLE
config migration to a new machine

### DISCUSSION THREAD

**Discussion by jackieli-tes** [neoclide/coc.nvim#3770] [Q&A] [answered]
I use [dotbot](https://github.com/anishathalye/dotbot) to manage all of the configurations. I manage the CocConfig as well as the coc extensions in it. Recently I'm moving all the configs to a new machine, but things just don't work smoothly

I have the whole `~/.config/coc` as well as the `~/.config/nvim/coc-settings.json` linked in my `~/.dotfiles` directory, and with `.gitignore` ignoring `**/node_modules/*` and `coc/extensions/coc-*-data/`. But in the target machine, it didn't just work out of the box. I tried cleaning and `yarn` to reinstall everything and some of them still didn't really work. I eventually just re-installed all the extensions from scratch.

I wonder is there a better or recommended way of doing this? Similar to the [kickstart.nvim](https://github.com/nvim-lua/kickstart.nvim) that bootstraps all the plugins : https://github.com/nvim-lua/kickstart.nvim/blob/master/init.lua#L2-L6

**Comment by chemzqm** [comment]
You should not copy `coc-*-data` which are created by extensions, they're not configuration and can be different on different machines

**Reply by jackieli-tes** [reply]
Yes I have those ignored in my .gitignore:

```
**/node_modules/*
coc/extensions/coc-*-data/
coc/mru
nvim/plugged
```

**Comment by jackielii** [comment]
I ended up adding `$HOME/.config/coc/extensions/package.json` to git and use `yarn install` in the `$HOME/.config/coc/extensions` folder to bootstrap: https://github.com/jackielii/dotfiles/blob/main/.config/yadm/bootstrap#L92

It works for most of the plugins. For now, I only found `coc-pyright` needed re-installing

**Comment by jackielii** [comment, marked as answer]
OK, now officially supported: https://github.com/neoclide/coc.nvim/discussions/3000

```
:h g:coc_global_extensions
```
